### PR TITLE
MINOR: follow-up to KAFKA-2464 for renaming/cleanup

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

**Comment by hachikuji** [comment]
@guozhangwang @becketqin @ewencp I did a little renaming as suggested by Becket. I decided to change the callback names used in AbstractCoordinator to reflect the current usage. As such, `onLeave` was changed to `onJoinPrepare`, and `onJoin` was changed to `onJoinComplete`. We can probably consider separately whether these are the semantics we want to have, but for now it seems best to make the names consistent with the implementation.

**Comment by becketqin** [comment]
@hachikuji Thanks. I think the names look much better now. I agree with your comments in #165 that it would be great if we can have the methods in rebalance listener to be consistent. Do you also want to change the name in ConsumerRebalanceListener? If we decide to do so, we'd better do this now than after 0.9 release.

@guozhangwang @ewencp What do you think?

**Comment by ewencp** [comment]
LGTM too.

Re: RebalanceListener, I just find the current behavior confusing because you get onPartitionsRevoked before you are even a member of the group, right? I think renaming could help. In the distributed Copycat patch, the equivalent just has different semantics -- the WorkerRebalanceListener.onRevoked method is only invoked if you actually _owned_ something that needed to be revoked. 

I think in the consumer we've just been leaning on the the fact that the empty assignment list from the subscriptions generally makes this work ok (whereas in Copycat you'd get a null pointer exception if you hadn't gone through the join process yet).

In any case, that shouldn't block this patch, just some additional thoughts on it since these are all related. That said, I think the ConsumerRebalanceListener semantics are more important to get right since it's public interface.

**Comment by hachikuji** [comment]
@becketqin @ewencp Yep, I'm on board with changing the callback names in ConsumerRebalanceListener (I also wonder if we can drop the 'Consumer' as well?). Would it make more sense to do that in a patch to KAFKA-2674 than here?

**Comment by guozhangwang** [comment]
Re: RebalanceListener, I would prefer the current function names just as @becketqin pointed out, since they are belonging to the Kafka consumer that are exposed to users who are mostly interested in the assigned partitions only, while `AbstractAssignor` and `AbstractCoordinator` APIs are exposed to programmers which are interested in general coordinator protocol. So changing the former's function names to be related to `Join` might confuse users who are not familiar with the protocol itself.

**Comment by hachikuji** [comment]
@guozhangwang I agree that the RebalanceListener should probably not refer directly to joining. That said, it would be nice to have names which suggested the actual semantics that are implemented so that we don't have issues like KAFKA-2674. Maybe we could rename `onPartitionsRevoked` to `onRebalancePrepare`, and `onPartitionsAssigned` to `onRebalanceComplete`? That would be consistent with this patch without introducing any unnecessary concepts to the user. In fact, since the interface name is already `RebalanceListener`, maybe we could make this even more concise and use `onPrepare` and `onComplete`.

**Comment by hachikuji** [comment]
@guozhangwang Rebased for Ewen's copycat patch. If the changes seem reasonable, maybe we can commit this and address RebalanceListener in KAFKA-2674?

**Comment by guozhangwang** [comment]
My concern with `onPrepare` is that it takes the revoked partitions as its parameter which is a bit confusing with the function name itself.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java [hunk 1]
```diff
@@ -295,7 +296,7 @@ public void onFailure(RuntimeException e) {
      * be polled to see if the request completed successfully.
      * @return A request future whose completion indicates the result of the JoinGroup request.
      */
-    private RequestFuture<ByteBuffer> sendJoinGroupRequest() {
+    private RequestFuture<ByteBuffer> performGroupJoin() {
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java]: Why do we want to change this function name? I feel the original function name is fine. Anyways, if we do want to change it we need to change the comments as well.

Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java]: The function does both JoinGroup and SyncGroup, so the name seemed a little misleading. Maybe I can be explicit and call it `performJoinAndSync`? Or maybe `joinNextGeneration`?

Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java]: Got it, then new name then looks good to me. Could you update the comments?

**Comment by guozhangwang** [comment]
@hachikuji Could you double check if this patch is compatible with the latest trunk as we have some recent merges that depend on this API which may not actually result in conflicts detectable by Github.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java [hunk 2]
```diff
@@ -209,7 +210,7 @@ public void ensureActiveGroup() {
 
         // onLeave only invoked if we have a valid current generation
         if (needsOnLeave) {
```
Comment (becketqin) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java]: Do we want to change this comments and variable name as well?

Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java]: Definitely. Thanks for catching it.

**Comment by hachikuji** [comment]
@guozhangwang Should be good to go. Builds clean locally after rebasing.

**Pull request closed without merging**